### DISCUSSION TITLE
Undo / Redo in TextArea

### DISCUSSION THREAD

**Discussion by lomaharshana** [Textualize/textual#3925] [Ideas]
It would be great if we can have undo/redo in the text area. IMHO, for a long text, like code editing, this is a feature that should come before advanced things like syntax highlighting.

**Comment by davep** [comment]
Full undo/redo will be coming to the `TextArea`; core support for it already exists in the code and has from the very beginning of its development.

**Reply by lomaharshana** [reply]
@davep Thank you! Could you please let me know what kind of core support for undo/redo exists and how to use it? Any pointer would be greatly appreciated.

**Reply by davep** [reply]
I'm not the author of that widget so can't say.
